### ISSUE TITLE
Ember-cli-rails breaking asset precompile

### ISSUE THREAD

**Issue by jgadbois** [thoughtbot/ember-cli-rails#212]
I'm integrating ember-cli-rails into my rails app.  I have my ember app in a root folder called `v2`.  I also have a folder `app/assets/stylesheets/v2`.  

When ember-cli-rails is enabled this somehow breaks asset pre-compilation - instead of only precompiling manifest files, the pre-compiler is attempting to precompile each scss file individually, which breaks on any scss file relying on another for mixins or whatever.

If I precompile with SKIP_EMBER, then it works.

**Comment by jgadbois** [comment]
I just renamed my stylesheet folder, but this is still an issue

**Comment by seanpdoyle** [comment]
Thanks for opening this issue and bringing this to our attention.

Would you mind providing more information around how you have your application set up?

Would you be able to share snippets of code that demonstrate the behavior you'd expect to see?

Are you trying to embed assets (in this case, stylesheets) from Rails directly into the Ember application?

Are you embedding the Ember application into the Rails view [using the helpers described here](https://github.com/thoughtbot/ember-cli-rails#usage)?

**Comment by jgadbois** [comment]
Sure, I can give more info and dig up some code snippets, but quickly here's what I have

My layout calls the ember-cli javascript helper using `include_ember_script_tags` and styles are included using the asset pipeline style sheet helper `stylesheet_link_tag`.

My stylesheet manifest was `app/assets/stylesheets/v2/wt_ember.scss` and my ember cli application was in the `v2` folder at the root level of the rails app.

**Comment by olivierobert** [comment]
+1 having the same issue as well. Similarly to @jgadbois I am relying on the asset pipeline style sheet helper for the ember application. This issue only concerns the compilation of stylesheets, JS files are fine.

**Comment by seanpdoyle** [comment]
@jgadbois @olivierobert could you please share some code snippets and error messages from this issue?

**Comment by jgadbois** [comment]
Layout
```
<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <%= stylesheet_link_tag "v2/wt_ember" %>
  </head>
  <body id='ember'>
    <%= javascript_include_tag( "weighttraining") %>
    <%= include_ember_script_tags :v2 %>
  </body>
</html>
```

production.rb
```
  config.assets.precompile += %w( v2/wt_ember.css ) #among others
```

Ember-CLI app location 
`<root>/v2`

Error
```
Sass::SyntaxError: Undefined mixin 'circle-border'.
```

It somehow is messing up all my includes and/or requires so mixins are seen as missing.  This problem goes away when renaming the v2 directory.

**Comment by seanpdoyle** [comment]
is the `circle-border` mixin defined in Rails' SCSS or Ember's? 

Which side does the Sass::Syntax error occur on: Rails or Ember?

**Comment by jgadbois** [comment]
There is no css or scss in the ember-cli app.  This is a migration from ember-rails and I so far have left all css in the Asset Pipeline.

Error occurs when precompiling assets for production, not during a request.  It's been a while since I did the rename, but I believe it all worked in the development environment.

**Comment by seanpdoyle** [comment]
Thanks for the context @jgadbois I'll look into this

**Comment by jgadbois** [comment]
Thanks!

**Comment by seanpdoyle** [comment]
@jgadbois @olivierobert would you mind checking out

**Comment by jgadbois** [comment]
@seanpdoyle in my case it's related to imports.  Possibly only imports from scss included from a Gem (like bootstrap or foundation).

@olivierobert do you have more information on your error if it's something similar?

I'll try to expand your sample app a bit to reproduce if I get some time later this week.

**Comment by seanpdoyle** [comment]
@jgadbois Has there been any movement on reproducing this in a simple example repo?

**Comment by jgadbois** [comment]
@seanpdoyle added some code to reproduce but when running specs I get

```
  3) User views ember app with Rails assets issue#212
     Failure/Error: visit page_path("issue_212")
     RuntimeError:
       EmberCLI app dependencies are not installed. From your Rails application root please run:

         $ bundle exec rake ember:install

       If you do not require Ember at this URL, you can restrict this check using the `enable`
       option in the EmberCLI initializer.
```

I don't actually see a dummy ember-cli app.

**Comment by seanpdoyle** [comment]
@jgadbois if you've cloned the repo, be sure to run `bin/setup`.

It'll download https://github.com/kellyselden/ember-cli-output into `spec/dummy/my-app`.

**Comment by seanpdoyle** [comment]
@jgadbois https://github.com/thoughtbot/ember-cli-rails/blob/master/CONTRIBUTING.md#setup

**Comment by jgadbois** [comment]
@seanpdoyle I attempted to reproduce the settings in my application as close as I could remember but it is still working for me.

https://github.com/thoughtbot/ember-cli-rails/pull/258

@olivierobert any other ideas?

**Comment by seanpdoyle** [comment]
@jgadbois @olivierobert 

There's been a lot of movement on this repo recently.

Could you try this again for version [0.5.3](https://rubygems.org/gems/ember-cli-rails/versions/0.5.3)?

**Comment by seanpdoyle** [comment]
I'm closing this issue due to its age and lack of reproducibility.

@jgadbois @olivierobert if you're able to reproduce this, please share whatever information you can find and I'll reopen.

**Comment by olivierobert** [comment]
@seanpdoyle I will try to replicate the issue on a more recent version as this was happening on v0.3.5

**Comment by jgadbois** [comment]
@seanpdoyle thanks for looking into this - sorry I wasn't able to reproduce.

**Comment by seanpdoyle** [comment]
Would you mind trying out:

* https://github.com/thoughtbot/ember-cli-rails/tree/remove-sprockets for `ember-cli-rails` gem
* https://github.com/rondale-sc/ember-cli-rails-addon/tree/remove-sprockets for `ember-cli-rails-addon`

Each of those branches has up-to-date README's.

---

I'm hopeful that we can get this fix landed and stabilize things for you.

Having said that, we don't plan on supporting the `include_ember_*_tags` helpers in post-`1.0.0` versions of this gem.

It is likely they'll be extracted to a stable gem, but I highly recommend you migrate toward using the `render_ember_app` helper, which relies 100% on EmberCLI to generate the page's HTML.

---

The second commit of https://github.com/seanpdoyle/ember-cli-rails-heroku-example/tree/remove-sprockets

Contains the entirety configuring an `include_ember_*_tags` application. 

The Ember app in question is `:admin` in that commit.

It currently powers https://ember-cli-rails-heroku-example.herokuapp.com/admin/

**Comment by jgadbois** [comment]
@seanpdoyle we have a hybrid app, so removing the tag helpers would definitely be a big negative for me.

**Comment by jgadbois** [comment]
It seems like my hybrid case would be common for this gem.  If completely rendering with Ember, I'd just have an ember-cli app that called a Rails API application.

**Comment by seanpdoyle** [comment]
@jgadbois thanks for the feedback!

I'm not sure what you mean by "hybrid" app, but if you mean embedding markup from Rails into an Ember app, that's achievable with the `render_ember_app` helper.

To learn more, read [Overriding the view](https://github.com/thoughtbot/ember-cli-rails/blob/remove-sprockets/README.md#overriding-the-view).

There are numerous issues with the `include_ember_*_helpers`.

1.Addons like [ember-cli-stripe](https://github.com/sweettooth/ember-cli-stripe) that use `contentFor` to [inject markup **required** by the addon into the `<head>`](https://github.com/sweettooth/ember-cli-stripe/blob/b466741e1a2bb57634b02a34ff33f844829f9f20/index.js#L6-L10), no longer work, since the `include_ember_*_tags` know nothing about the EmberCLI-generated `index.html`. This breaks the addon (and others like it)
1. they're created by the gem on the fly. Determining which assets to serve is somewhat complicated once we they're fingerprinted (i.e. in `production`). They end up being maintenance overhead we'd like to avoid.
1. We don't want to get into the business of reimplementing our own version of EmberCLI functionality, or diverging from EmberCLI's behavior. We'd rather depend on and existing behaviors.

**TL;DR**

Before cutting a `1.0.0` that drops support for `include_ember_*_tags` helpers, it is likely we'll extract a gem that contains those helpers and the necessary code to support them.

This functionality won't vanish, but it won't be supported in the future (at least not as a part of this gem).

Migrating to use `render_ember_app` is the recommended move, and its probably just as flexible as using the `include_ember_*_tags` helpers for your particular use case.

**Comment by jgadbois** [comment]
@seanpdoyle that makes sense I think.  By hybrid I just mean that some of our pages our completely rendered by Rails (most of our public facing pages), but our dashboard pages are all powered by Ember (but within the same Rails layout).

So for us it makes sense to have the header/footer/css always controlled by Rails and the Asset Pipeline.

I think with overrides it looks like that will still work if I am reading it correctly.

**Comment by seanpdoyle** [comment]
@jgadbois correct.

`render_ember_app` still supports injecting any arbitrary markup managed by your rails app (HTML tags, additional `<script>` tags, additional stylesheet `<link>` tags, etc).
